libs/langgraph/src/graph/graph.ts

```diff
@@ -29,22 +29,32 @@ import {
 } from "../constants.js";
 import { gatherIteratorSync, RunnableCallable } from "../utils.js";
 import { InvalidUpdateError, NodeInterrupt } from "../errors.js";
+import { StateDefinition, StateType } from "./annotation.js";
+import type { LangGraphRunnableConfig } from "../pregel/runnable_types.js";
 
 /** Special reserved node name denoting the start of a graph. */
 export const START = "__start__";
 /** Special reserved node name denoting the end of a graph. */
 export const END = "__end__";
 
-export interface BranchOptions<IO, N extends string> {
+export interface BranchOptions<
+  IO,
+  N extends string,
+  CallOptions extends LangGraphRunnableConfig = LangGraphRunnableConfig
+> {
   source: N;
-  path: Branch<IO, N>["condition"];
+  path: Branch<IO, N, CallOptions>["condition"];
   pathMap?: Record<string, N | typeof END> | (N | typeof END)[];
 }
 
-export class Branch<IO, N extends string> {
+export class Branch<
+  IO,
+  N extends string,
+  CallOptions extends LangGraphRunnableConfig = LangGraphRunnableConfig
+> {
   condition: (
     input: IO,
-    config?: RunnableConfig
+    config: CallOptions
```

Comment:
Now it's a required arg?

Reply:
Ah this is an internal function, ignore me

Reply:
I usually do!